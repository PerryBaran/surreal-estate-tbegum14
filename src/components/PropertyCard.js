import React from "react";
import "../styles/property-card.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
	faBath,
	faBed,
	faSterlingSign,
	faEnvelope,
} from "@fortawesome/free-solid-svg-icons";

export default function PropertyCard(props) {
	const { title, price, bathrooms, bedrooms, type, city, email } = props;
	return (
		<div className="property-card">
			<div>
				<h3 className="property-card_title">{title}</h3>
				<p className="property-card_city">
					{type} - {city}
				</p>
				<p className="property-card_bedrooms">
					<FontAwesomeIcon icon={faBed} /> {bedrooms}
				</p>
				<p className="property-card_bathrooms">
					<FontAwesomeIcon icon={faBath} /> {bathrooms}
				</p>
				<p className="property-card_price">
					<FontAwesomeIcon icon={faSterlingSign} /> {price}
				</p>
			</div>

			<a href={email}>
				<button className="property-card_button" type="button">
					<FontAwesomeIcon icon={faEnvelope} />   Email
				</button>
			</a>
		</div>
	);
}
