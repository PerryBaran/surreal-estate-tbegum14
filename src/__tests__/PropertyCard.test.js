import React from "react";
import {render, screen} from "@testing-library/react"
import PropertyCard from "../components/PropertyCard"

describe("property card", ()=>{
    const validProps = {
        title: "property title",
        type: "property type",
        bedrooms: 0,
        bathrooms: 0,
        price: 0,
        city: "Manchester",
        email: "",
    }
    it("renders properly", ()=>{
        const {asFragment} = render(<PropertyCard />)

        expect(asFragment).toMatchSnapshot()
    })

    it("renders with the correct title", ()=>{
        render(<PropertyCard title={validProps.title} />)
        const linkElement = screen.getByText("property title", {exact: true})
        expect(linkElement).toBeInTheDocument()
    })
})