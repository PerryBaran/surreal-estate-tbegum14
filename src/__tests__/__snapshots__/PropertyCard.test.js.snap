// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`property card renders properly 1`] = `[Function]`;
